app/uk/gov/hmrc/digitalservicestax/controllers/ReturnsController.scala

```diff
@@ -92,74 +83,67 @@ class ReturnsController @Inject()(
 
   def showAmendments(): Action[AnyContent] = authorisedAction.async {
     implicit request: AuthorisedRequest[AnyContent] =>
-      implicit val msg: UniformMessages[Html] = messages(request)
-
-       implicit val persistence: MongoPersistence[AuthorisedRequest[AnyContent]] =
-         MongoPersistence[AuthorisedRequest[AnyContent]](
-           mongo,
-           collectionName = "uf-amendments-returns",
-           appConfig.mongoJourneyStoreExpireAfter
-         )(_.internalId)
+    implicit val msg: UniformMessages[Html] = messages(request)
 
-      backend.lookupRegistration().flatMap{
-        case None      => Future.successful(NotFound)
-        case Some(_) => Future.successful(NotFound)
-          backend.lookupAmendableReturns().map { outstandingPeriods =>
-             outstandingPeriods.toList match {
-               case Nil =>
-                NotFound
-               case periods =>
-                Ok(layout(
-                    pageTitle =
-                      Some(s"${msg("resubmit-a-return.title")} - ${msg("common.title")} - ${msg("common.title.suffix")}")
-                )(resubmitAReturn("resubmit-a-return", periods, periodForm)(msg, request)))
-             }
-            }
+    backend.lookupRegistration().flatMap{
+      case None      => Future.successful(NotFound)
+      case Some(_) => Future.successful(NotFound)
```

Comment:
What is the reason of this `Future.successful(NotFound)`.

You have implementation for this case below so this element is redundant.

Reply:
I'm agree, but this code hasn't changed apart from the indentation and I don't want to expand the scope of the PR to open-ended refactoring. We can always create a new ticket if needed.